src/actions/graph.js

```diff
@@ -39,21 +79,46 @@ const chooseLayout = (instance, state) =>
 // Prepare new layout worker
 const layoutWorker = preventWorkerQueues(worker, chooseLayout);
 
+/**
+ * Formula to estimate the point at which the graph will take unreasonably
+ * long to render in terms of input nodes and edges.
+ *  @param {Object} largeGraphThreshold The defined threshold for a large graph
+ *  @param {integer} nodeCount The amount of nodes to be displayed in the flowchart
+ *  @param {integer} edgeCount The amount of edges to be displayed in the flowchart
+ */
+const isLarge = (largeGraphThreshold, nodeCount, edgeCount) => {
+  return nodeCount + 1.5 * edgeCount > largeGraphThreshold ? true : false;
+};
+
 /**
  * Async action to calculate graph layout in a web worker
  * whiled displaying a loading spinner
  * @param {Object} graphState A subset of main state
  * @return {function} A promise that resolves when the calcuation is done
  */
-export function calculateGraph(graphState) {
+export function calculateGraph(
+  graphState,
+  displayThreshold = largeGraphThreshold
+) {
   if (!graphState) {
     return updateGraph(graphState);
   }
   return async function(dispatch) {
-    dispatch(toggleLoading(true));
-    const graph = await layoutWorker(graphState);
-    dispatch(toggleGraph(true));
-    dispatch(toggleLoading(false));
-    return dispatch(updateGraph(graph));
+    const { nodes, edges, displayLargeGraph } = graphState;
+
+    if (
+      isLarge(displayThreshold, nodes.length, edges.length) === true &&
+      displayLargeGraph === false
+    ) {
+      dispatch(toggleIsLarge(true));
```

Comment:
As below, this action could also probably just be a selector instead. You don't need to trigger a change in the state to do this - just `return`.

Reply:
Thanks Richard! I had just pushed the refactor to set this as a selector in the latest commit ( the pairing we did earlier), let me know if you have any further comments! 

Resolving this issue as this is now deleted.